client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -261,6 +261,32 @@ public synchronized int getId(String subject, Schema schema)
     }
   }
 
+  @Override
+  public List<Integer> deleteSubject(String subject) throws IOException, RestClientException {
+    return deleteSubject(restService.DEFAULT_REQUEST_PROPERTIES, subject);
+  }
+
+  @Override
+  public List<Integer> deleteSubject(Map<String, String> requestProperties, String subject)
```

Comment:
@hjafarpour Sorry, missed this in my first pass. We will have to remove these from the in-memory DS as well here. We should be able to look up the schema by the id. But its ok to remove from there as well if its easier. We can probably do that before calling the REST service. The reason is if tis not there in cache worst case, we go to server vs if server successfully deleted but cache update fails.